viewer/packages/cad-styling/src/NodeIdNodeCollection.ts

```diff
@@ -0,0 +1,89 @@
+/*!
+ * Copyright 2022 Cognite AS
+ */
+import { CogniteClient, ListResponse, Node3D } from '@cognite/sdk';
+import chunk from 'lodash/chunk';
+import { CdfModelNodeCollectionDataProvider } from './CdfModelNodeCollectionDataProvider';
+import { CdfNodeCollectionBase } from './CdfNodeCollectionBase';
+import { NodeCollectionSerializationContext, registerNodeCollectionType } from './NodeCollectionDeserializer';
+import { SerializedNodeCollection } from './SerializedNodeCollection';
+
+/**
+ * Collection that holds a set of nodes including children identified by nodeIds. Note that
+ * this involves remapping IDs to "tree indices" and subtree sized used by Reveal using
+ * the CDF API. Its often better to use one of the other node collections or {@see TreeIndexNodeCollection}
+ * whenever possible for best performance.
+ */
+export class NodeIdNodeCollection extends CdfNodeCollectionBase {
+  public static readonly classToken = 'NodeIdNodeCollection';
+
+  private readonly _client: CogniteClient;
+  private readonly _model: CdfModelNodeCollectionDataProvider;
+
+  private _nodeIds: number[] = [];
+
+  constructor(client: CogniteClient, model: CdfModelNodeCollectionDataProvider) {
+    super(NodeIdNodeCollection.classToken, model);
+    this._client = client;
+    this._model = model;
+  }
+
+  /**
+   * Populates the collection with the nodes with the IDs provided. All children of
```

Comment:
```suggestion
   * Populates the collection with the nodes from the IDs provided. All children of
```

Reply:
Hmm, to me it makes more sense with 'with'. @christjt